extensions/amp-story/1.0/amp-story.js

```diff
@@ -1619,11 +1619,14 @@ export class AmpStory extends AMP.BaseElement {
     if (this.win.MutationObserver) {
       if (!this.sidebarObserver_) {
         this.sidebarObserver_ = new this.win.MutationObserver(mutationsList => {
-          if (mutationsList.some(
-              mutation => mutation.attributeName === 'open')) {
-            this.storeService_.dispatch(Action.TOGGLE_SIDEBAR,
-                this.sidebar_.hasAttribute('open'));
-          }
+          mutationsList.forEach(mutation => {
+            if (mutation.attributeName === 'open') {
+              this.storeService_.dispatch(Action.TOGGLE_SIDEBAR,
+                  this.sidebar_.hasAttribute('open'));
+            } else if (mutation.attributeName === 'hidden') {
+              this.storeService_.dispatch(Action.TOGGLE_SIDEBAR, false);
+            }
```

Comment:
So, I'm sorry I should have checked all this more closely earlier.

When the sidebar gets closed, the `open` attribute is removed, and the `hidden` attribute is added. So if you follow the execution of this code block, the `SIDEBAR_STATE` will be set to false when the `open` attribute is removed. You actually don't need any code change here.